### PR TITLE
Add java sol to closing the farm (silver)

### PR DESCRIPTION
**Pull request by nathangong**

### CODE REVIEW

File: solutions/USACO open16-closing-the-farm.mdx [hunk 1]
```diff
@@ -15,6 +15,10 @@ each barn, we can store a boolean array and set each barn in the iteration to be
 If all the unclosed nodes have been visited, then we print "YES"; otherwise, we
 print "NO".
 
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:20]:
```suggestion
**Time Complexity:** $\mathcal{O}(N^2 + NM)$
```

File: solutions/USACO open16-closing-the-farm.mdx [hunk 2]
```diff
@@ -170,4 +174,70 @@ void dfs(int barn)
 
 </CPPSection>
 
+<JavaSection>
+
+```java
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:179]: use tabs instead of spaces to indent

File: solutions/USACO open16-closing-the-farm.mdx [hunk 3]
```diff
@@ -170,4 +174,70 @@ void dfs(int barn)
...
+            }
+            closed[order[i]] = true;
+        }
+        out.close();
+    }
+
+    static void dfs(int node) {
+        if (visited[node] || closed[node]) return;
+
+        // Increase the number of nodes if and only if they haven't been closed or visited previously
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:231]: this line is too long

File: solutions/USACO open16-closing-the-farm.mdx [hunk 4]
```diff
@@ -170,4 +174,70 @@ void dfs(int barn)
...
+        int m = sc.nextInt();
+
+        visited = new boolean[n];
+        closed = new boolean[n];
+        adj = new ArrayList<ArrayList<Integer>>();
+        for (int i = 0; i < n; i++) adj.add(new ArrayList<Integer>());
+
+        for (int i = 0; i < m; i++) {
+            int a = sc.nextInt()-1;
+            int b = sc.nextInt()-1;
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:203]:
```suggestion
            int a = sc.nextInt() - 1;
            int b = sc.nextInt() - 1;
```

File: solutions/USACO open16-closing-the-farm.mdx [hunk 5]
```diff
@@ -170,4 +174,70 @@ void dfs(int barn)
...
+            adj.get(a).add(b);
+            adj.get(b).add(a);
+        }
+
+        int[] order = new int[n];
+        for (int i = 0; i < n; i++) {
+            order[i] = sc.nextInt()-1;
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:210]:
```suggestion
            order[i] = sc.nextInt() - 1;
```

File: solutions/USACO open16-closing-the-farm.mdx [hunk 6]
```diff
@@ -170,4 +174,70 @@ void dfs(int barn)
...
+        }
+
+        for (int i = 0; i < n; i++) {
+            Arrays.fill(visited, false);
+            nodes = 0;
+            
+            dfs(order[n-1]); // DFS from the barn that will close last
+            if (nodes == n-i) { // Checks if all unclosed barns have been visited
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:218]:
```suggestion
            dfs(order[n - 1]); // DFS from the barn that will close last
            // Checks if all unclosed barns have been visited
            if (nodes == n - i) {
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/USACO open16-closing-the-farm.mdx [hunk 7]
```diff
@@ -170,4 +174,71 @@ void dfs(int barn)
...
+            }
+            closed[order[i]] = true;
+        }
+        out.close();
+    }
+
+    static void dfs(int node) {
+        if (visited[node] || closed[node]) return;
+
+        // Increase the number of nodes iff they haven't already been closed or visited
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:232]:
```suggestion
        // Increase the number of nodes 
        // iff this barn hasn't already been closed or visited
```
still too long...maybe split it into two lines like this

File: solutions/USACO open16-closing-the-farm.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/USACO open16-closing-the-farm.mdx:179]: Fixed this just now, can you check if it's right?

**Comment by maggie-j-liu** [comment]
@jessechoe10 do you want to detemplate your solution?

File: solutions/USACO open16-closing-the-farm.mdx (hunk 2 above)
Comment (maggie-j-liu) [reply solutions/USACO open16-closing-the-farm.mdx:179]: Yep, it looks good now.

File: solutions/USACO open16-closing-the-farm.mdx [hunk 8]
```diff
@@ -170,4 +174,72 @@ void dfs(int barn)
...
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class closing {
+	//Store global variables used by dfs
```
Comment (maggie-j-liu) [inline solutions/USACO open16-closing-the-farm.mdx:184]:
```suggestion
	// Store global variables used by dfs
```
to be consistent with the other comments

**Review by maggie-j-liu — APPROVED** [review]

**Review by frodakcin — APPROVED** [review]

**Comment by frodakcin** [comment]
> @jessechoe10 do you want to detemplate your solution?

Oops, did not see this; make new pr?

**Pull request merged**